### PR TITLE
Adds merge sort done without recursion

### PR DESCRIPTION
**Pull request by TrevorJamesH**

### CODE REVIEW

File: src/mergeSortLoop.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
+export default function mergeSort(toSort){
+  let currentLevel = []
```
Comment (jrobcodes) [inline src/mergeSortLoop.js:2]: `.map` returns the collection created by applying the function specified as the parameter to each element in the original collection.  This could be leveraged to make lines 2-7 more concise:
```js
const currentLevel = toSort.map( number => [ number ] )
```

File: src/mergeSortLoop.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  toSort.map( function(number){
+    let subArray = []
+    subArray.push(number)
+    currentLevel.push(subArray)
+  })
+  let nextLevel = []
+
+  while(currentLevel.length > 1){
```
Comment (jrobcodes) [inline src/mergeSortLoop.js:10]: There's a lot going on in this loop - consider refactoring to pull out logic that can exist in isolation (for example, the swapping of elements between sub arrays could be placed in a separate function).

File: test/mergeSortLoop.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import { expect } from 'chai'
+import mergeSort from '../src/mergeSort'
+
+describe.only('mergeSort()', function(){
```
Comment (jrobcodes) [inline test/mergeSortLoop.js:4]: When defining your test suite, you should also consider edge cases: how should merge sort behave with an empty array?  how should merge sort behave with a non-array (a string, collection, null, undefined, etc.)
